src/charm.py

```diff
@@ -214,6 +216,9 @@ def _validate_config_params(self):
             raise ValueError(
                 f"config: invalid self-registration role {default_role!r}"
             )
+        for key, value in self.config.items():
+            if value == "":
```

Comment:
Won't this force users to include an `http_proxy` config, even if they don't need one?

Reply:
Nope, it checks for empty strings. The default value is and can remain `None`.

Reply:
Are we sure the default for config strings is `None` and not `""`?

Reply:
I'm fairly certain, have you seen a scenario when this is otherwise?

Reply:
@kelkawi-a I believe you are setting the default to `""` in your charms with this `default: ""` but if you do not set anything then I believe the value is `None`.

Reply:
I just did a quick test on a charm I have running by doing `juju config <charm_name> --reset <config_key>`, which reset the value back to `""`.

Reply:
Is the default value for that config paramter set to be `""` as I described above?

Reply:
Ah you're right. Nevermind me then.